internal/buildjob/buildjob_test.go

```diff
@@ -0,0 +1,235 @@
+			},
+		},
+	)
+
+	ctx := context.Background()
+	log := NewLogger()
+	job := Job{
+		log: log,
+	}
+
+	t.Run("bad_basic_auth_both_inline_and_secret", func(t *testing.T) {
+		apiRegs := []v1alpha1.Registry{
+			{
+				Server: "test.com",
+				BasicAuth: v1alpha1.BasicAuthConfig{
+					Username:        "this",
+					Password:        "should",
+					SecretName:      "never",
+					SecretNamespace: "happen",
+				},
+			},
+		}
+		_, err := job.buildRegistryConfigs(ctx, apiRegs)
+		assert.Error(t, err)
+	})
+
+	t.Run("no_auth_basic_auth", func(t *testing.T) {
+		apiRegs := []v1alpha1.Registry{noAuthRegistryConfig}
+		registries, err := job.buildRegistryConfigs(ctx, apiRegs)
+
+		assert.NoError(t, err)
+		assert.ElementsMatch(t, registries, []config.Registry{noAuthRegistry})
+	})
+
+	t.Run("single_basic_auth_from_inline", func(t *testing.T) {
+		apiRegs := []v1alpha1.Registry{inlineRegistryConfig}
+		registries, err := job.buildRegistryConfigs(ctx, apiRegs)
+
+		assert.NoError(t, err)
+		assert.ElementsMatch(t, registries, []config.Registry{inlineRegistry})
+	})
+
+	t.Run("single_basic_auth_from_inline_with_nonssl", func(t *testing.T) {
+		inlineRegistryConfigWithNonSSl := inlineRegistryConfig
+		inlineRegistryConfigWithNonSSl.NonSSL = true
+
+		inlineRegistryWithNonSSl := inlineRegistry
+		inlineRegistryWithNonSSl.NonSSL = true
+
+		apiRegs := []v1alpha1.Registry{inlineRegistryConfigWithNonSSl}
+		registries, err := job.buildRegistryConfigs(ctx, apiRegs)
+
+		assert.NoError(t, err)
+		assert.ElementsMatch(t, registries, []config.Registry{inlineRegistryWithNonSSl})
+	})
+
+	t.Run("single_basic_auth_from_secret", func(t *testing.T) {
+		job.clientk8s = clientWithSingleRegistrySecret
+
+		apiRegs := []v1alpha1.Registry{singleRegistrySecretRegistryConfig}
+		registries, err := job.buildRegistryConfigs(ctx, apiRegs)
+
+		assert.NoError(t, err)
+		assert.ElementsMatch(t, registries, []config.Registry{secretRegistry1})
+	})
+
+	t.Run("load_implicit_basic_auth_from_secret", func(t *testing.T) {
+		job.clientk8s = clientWithTwoRegistriesSecret
+
+		apiRegs := []v1alpha1.Registry{twoRegistriesSecretRegistryConfig}
+		registries, err := job.buildRegistryConfigs(ctx, apiRegs)
+
+		assert.NoError(t, err)
+		assert.ElementsMatch(t, registries, []config.Registry{secretRegistry1, secretRegistry2})
+	})
+
+	t.Run("override_implicit_basic_auth_from_secret", func(t *testing.T) {
+		job.clientk8s = clientWithTwoRegistriesSecret
+
+		apiRegs := []v1alpha1.Registry{twoRegistriesSecretRegistryConfig, inlineRegistryConfig}
+		registries, err := job.buildRegistryConfigs(ctx, apiRegs)
+
+		assert.NoError(t, err)
+		assert.ElementsMatch(t, registries, []config.Registry{secretRegistry1, inlineRegistry})
+
+		// Test order irrelevant to overriding
+		apiRegsReverseOrder := []v1alpha1.Registry{inlineRegistryConfig, twoRegistriesSecretRegistryConfig}
+		registriesReverseOrder, err := job.buildRegistryConfigs(ctx, apiRegsReverseOrder)
```

Comment:
so this is a bit tricky since I believe you are walking a map to build the registry list which has no order. Go will randomize this walk so it can fail. If assert.ElementsMatch does not care about order, we are good.

Reply:
why call it registriesReverseOrder if you cannot guarantee the order it comes back as?